src/handle.rs

```diff
@@ -151,6 +151,8 @@ impl<T> Global<T> {
   }
 }
 
+unsafe impl<T> Send for Global<T> {}
```

Comment:
I don't think Global can safely be sent between threads?

Reply:
I think there is no material difference between `Isolate` being `Send` and `Global<T>` being `Send`.
Any one of those already creates the possibility to access `Global` while the `Isolate` is accessed by another thread, which creates safety issues that need to be addressed.

The issue arises from calling the non-thread-safe `IsolateHandle::get_isolate_ptr()` method at these call sites:

* https://github.com/denoland/rusty_v8/blob/main/src/handle.rs#L337
* https://github.com/denoland/rusty_v8/blob/main/src/handle.rs#L273
* https://github.com/denoland/rusty_v8/blob/main/src/handle.rs#L164

I believe the only way to address this issue is to add a check to `get_isolate_ptr()` that verifies that the `Isolate` is locked by the current thread.

Reply:
Thanks for the insight, I think I have a better understanding of the handles now.. I attempted to resolve these safety issues in the latest commits.

It's unfortunate that slightly misusing a Global is a runtime assertion Specifically, dropping while outside the isolate scope.

However, as an embedder, storing and accessing Globals within the bounds of the Isolate's entered state (ie. using `set/get_slot`) largely prevents these occurrences (i think).